### PR TITLE
chore(schema): common java library for ETL

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
- [ ] update `sonar-project`. 1. exclude build and gradle for `etl-common`, 2. add test results to `sonar.coverage.jacoco.xmlReportPaths`

File: .projenrc.js [hunk 1]
```diff
@@ -277,6 +277,10 @@ const project = new awscdk.AwsCdkTypeScriptApp({
     'src/data-pipeline/spark-etl/build',
     'src/data-pipeline/spark-etl/bin',
     'src/data-pipeline/spark-etl/?/',
+    'src/data-pipeline/etl-common/.gradle',
+    'src/data-pipeline/etl-common/build',
+    'src/data-pipeline/etl-common/bin',
+    'src/data-pipeline/etl-common/?/',
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply .projenrc.js:283]: removed`/?/'`

File: src/data-pipeline/etl-common/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Project
+group=software.aws.solution.common
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/data-pipeline/etl-common/gradle.properties:2]: yes.

File: src/common/s3-asset.ts [hunk 3]
```diff
@@ -37,10 +39,21 @@ export function uploadBuiltInJarsAndRemoteFiles(
   if (props.shadowJar) {
     jarFile = `${props.jarName}-${version}-all.jar`;
   }
+
+  let commonLibCommands = [
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/common/s3-asset.ts:43]: changed.

File: src/data-pipeline/etl-common/src/main/java/software/aws/solution/clickstream/common/gtm/GTMEventParser.java [hunk 4]
```diff
@@ -227,22 +242,24 @@ private void setTrafficSource(final ClickstreamEvent clickstreamEvent) {
...
-            clickstreamEvent.setTrafficSourceContent(trafficSource.getContent());
-            clickstreamEvent.setTrafficSourceMedium(trafficSource.getMedium());
-            clickstreamEvent.setTrafficSourceSource(trafficSource.getSource());
-            clickstreamEvent.setTrafficSourceTerm(trafficSource.getTerm());
-            clickstreamEvent.setTrafficSourceClid(trafficSource.getClid());
-            clickstreamEvent.setTrafficSourceClidPlatform(trafficSource.getClidPlatform());
-            clickstreamEvent.setTrafficSourceCampaignId(trafficSource.getCampaignId());
-            clickstreamEvent.setTrafficSourceChannelGroup(trafficSource.getChannelGroup());
-            clickstreamEvent.setTrafficSourceCategory(trafficSource.getCategory());
+            CategoryTrafficSource cTrafficSource = parserResult.getTrafficSource();
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/data-pipeline/etl-common/src/main/java/software/aws/solution/clickstream/common/gtm/GTMEventParser.java]: changed

File: src/data-pipeline/etl-common/settings.gradle [hunk 5]
```diff
@@ -0,0 +1,2 @@
+rootProject.name = 'etl-common'
+//include('app')
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/data-pipeline/etl-common/settings.gradle]: removed

**Review by zxkane — APPROVED** [review]

**Pull request merged**